src/coap_blockwise.c

```diff
@@ -22,13 +22,144 @@ struct blockwise_transfer
+}
+
+// Function to call the application's read block callback after a successful
+// blockwise upload
+static void call_read_block_callback(struct blockwise_transfer *ctx)
+{
+    if (ctx->callback.read_cb(ctx->offset,
+                              ctx->block_buffer,
+                              &ctx->block_size,
+                              &ctx->is_last,
+                              ctx->callback_arg)
+        != GOLIOTH_OK)
+    {
+        // TODO: handle application callback error
+    }
+}
+
+// Function to upload a single block
+static enum golioth_status upload_single_block(struct golioth_client *client,
+                                               struct blockwise_transfer *ctx)
+{
+    assert(ctx->block_size > 0 && ctx->block_size <= CONFIG_GOLIOTH_BLOCKWISE_UPLOAD_BLOCK_SIZE);
+    return golioth_coap_client_set_block(client,
+                                         ctx->path_prefix,
+                                         ctx->path,
+                                         ctx->is_last,
+                                         GOLIOTH_CONTENT_TYPE_JSON,
+                                         ctx->offset,
+                                         ctx->block_buffer,
+                                         ctx->block_size,
+                                         on_block_sent,
+                                         ctx,
+                                         true,
+                                         GOLIOTH_SYS_WAIT_FOREVER);
+}
+
+// Function to handle blockwise upload errors
+static enum golioth_status handle_upload_error()
+{
+    // TODO: handle errors like disconnects etc
+    return GOLIOTH_OK;
+}
+
+// Function to manage blockwise upload and handle errors
+static enum golioth_status process_blockwise_uploads(struct golioth_client *client,
+                                                     struct blockwise_transfer *ctx)
+{
+    enum golioth_status status = GOLIOTH_ERR_FAIL;
+    ctx->block_size = 0;
+    call_read_block_callback(ctx);
+    if (ctx->block_size)
+    {
+        status = upload_single_block(client, ctx);
+        if (ctx->status == GOLIOTH_OK)
+        {
+            ctx->offset++;
+        }
+        else
+        {
+            // TODO: handle_upload_error
+            status = handle_upload_error();
+        }
+    }
+    return status;
+}
+
+enum golioth_status golioth_blockwise_post(struct golioth_client *client,
+                                           const char *path_prefix,
+                                           const char *path,
+                                           read_block_cb cb,
+                                           void *callback_arg)
+{
+    enum golioth_status status = GOLIOTH_ERR_FAIL;
+    assert(client);
+    assert(path);
+    assert(cb);
+
+    struct blockwise_transfer *ctx = malloc(sizeof(struct blockwise_transfer));
+    uint8_t *data_buff = malloc(CONFIG_GOLIOTH_BLOCKWISE_UPLOAD_BLOCK_SIZE);
+    assert(data_buff);
+    assert(ctx);
```

Comment:
Is it safe to use assert here rather than issuing an error message and returning? For instance, does this cause the program to crash if `malloc()` fails due to lack of memory?